api/decorators.py

```diff
@@ -0,0 +1,16 @@
+import functools
+
+from rest_framework.exceptions import ValidationError as RestValidationError
+from django.core.exceptions import ValidationError
+
+
+def rethrow_validation_error_for_serializer(func):
```

Comment:
Could you explain why this was necessary? I'm only partially following.

Reply:
exception in the serializer == HTTP 400, exception in model == HTTP 500